### DISCUSSION TITLE
Very slow clones for the last 3 weeks, 3Kbs

### DISCUSSION THREAD

**Discussion by mangelozzi** [community/community#12896] [Other Feature Feedback, Questions, & Ideas]
Firstly, you can see, my connection is fine, 48Mbs down, 47Mbs up. This is a speed test running while peforming a git clone:
![image](https://user-images.githubusercontent.com/19764314/158439623-be1bacd5-16ce-4122-9541-7b9dd57e73df.png)

And heres the very slow clone (3Kbs) , tried 3 times to checkout 25MB repo, took over an hour to checkout 12MB (used --depth=3).
![image](https://user-images.githubusercontent.com/19764314/158439598-50517023-1fe3-4f31-b0e4-2cecde5bd14e.png)

Using github debug (https://github-debug.com/):
It says post the results in support, and this category seemed like the best fit, so I don't know if its in the right place, but here the results:

```
Error downloading
2507998 bytes downloaded from cloud.githubusercontent.com at 6.14 Mbps
2507998 bytes downloaded from github.githubassets.com at 24.07 Mbps
2507998 bytes downloaded from sea.github-debug.com at 0.56 Mbps
2507998 bytes downloaded from iad.github-debug.com at 0.51 Mbps
Testing download from fra.github-debug.com...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...


               

169.0.96.104


               

Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36


               

78f86c8ee141c16c2265840f42f9ff12


               

9A88:CCC3:0CE1:2E664:6230CC30


               

               



 
traceroute to 169.0.96.104 (169.0.96.104), 30 hops max, 60 byte packets
 1  ec2346f690462de20549ef85180e1212  0.263 ms
 2  5148ad31bb6cdd25b4822c53432afb2a  0.165 ms
 3  3694fbc08d4c14f8747440556344f58b  0.349 ms
 4  e8bbacbac2822afa2b9334bbf001e5da  0.213 ms
 5  4cc7dda2ade2fd2b91e3e83705761648  0.658 ms
 6  ffm-b11-link.ip.twelve99.net (62.115.182.170)  0.877 ms
 7  ae1.cs1.fra9.de.eth.zayo.com (64.125.29.64)  0.968 ms
 8  ae27.mpr1.fra4.de.zip.zayo.com (64.125.30.255)  0.805 ms
 9  HundredGE0-3-0-0.br02.frf06.pccwbtn.net (63.218.232.33)  1.336 ms
10  *
11  *
12  *
13  *
14  *
15  *
16  *
17  *
18  *
19  *
20  *
21  *
22  *
23  *
24  *
25  *
26  *
27  *
28  *
29  *
30  *
 
PING 169.0.96.104 (169.0.96.104) 56(84) bytes of data.

--- 169.0.96.104 ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 1999ms

 
Start: 2022-03-15T10:26:35-0700
HOST: b8f837a32fb4b3a96f704188b43f67e0                           Loss%   Snt   Last   Avg  Best  Wrst StDev
  1.|-- ec2346f690462de20549ef85180e1212                            0.0%    10    0.3   0.3   0.2   0.4   0.1
  2.|-- fc7e6149a0e922bc65fabc76901eaa02                            0.0%    10    0.3   0.4   0.3   0.4   0.0
  3.|-- 26db9eaa194a4674cbf7c73e12787db8                            0.0%    10    0.6   1.4   0.4   8.2   2.4
  4.|-- 39ca010b3ed39fb6f8e4b8d0bb2e6582                            0.0%    10    0.4   0.4   0.3   0.4   0.0
  5.|-- 785f9def70ac3712d83179c2cf02066a                            0.0%    10   11.9   4.9   0.6  12.9   4.9
  6.|-- 0.et-1-0-2.mcs1.fra9.de.zip.zayo.com (82.98.193.28)         0.0%    10    0.9   1.1   0.9   1.6   0.2
  7.|-- ae1.cs1.fra9.de.eth.zayo.com (64.125.29.64)                 0.0%    10    1.1   1.2   1.1   1.4   0.1
  8.|-- ae27.mpr1.fra4.de.zip.zayo.com (64.125.30.255)              0.0%    10    1.3   1.8   0.9   4.0   1.0
  9.|-- 63-222-0-65.static.pccwglobal.net (63.222.0.65)             0.0%    10    1.7   2.0   1.5   3.1   0.6
 10.|-- HundredGE0-3-0-0.br02.frf06.pccwbtn.net (63.218.232.33)     0.0%    10    1.8   1.8   1.7   2.0   0.1
 11.|-- telkom.te0-1-0-15.br02.frf06.pccwbtn.net (63.218.233.102)  20.0%    10   55.8  55.7  55.5  56.0   0.1
 12.|-- ???                                                        100.0    10    0.0   0.0   0.0   0.0   0.0
 13.|-- ???                                                        100.0    10    0.0   0.0   0.0   0.0   0.0
 14.|-- ???                                                        100.0    10    0.0   0.0   0.0   0.0   0.0
 15.|-- ???                                                        100.0    10    0.0   0.0   0.0   0.0   0.0
 16.|-- 169-0-96-1.ip.afrihost.co.za (169.0.96.1)                  30.0%    10  228.2 228.3 228.2 228.5   0.1
 17.|-- ???                                                        100.0    10    0.0   0.0   0.0   0.0   0.0
 
104.96.0.169.in-addr.arpa domain name pointer 169-0-96-104.ip.afrihost.co.za.
```    

What has changed recently?

**Comment by illsaveus** [comment]
I'm having this same problem! If the repo is large enough, it literally never completes the download. Your internet is faster than mine but I have a similar line as your ??? line where 100% of the packets are lost for some reason. 

                
Testing download from github.com...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...
Please wait ...

                
72.134.113.196

                
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36

                
f46ac263fdfd95deec79ac0e884cb6fa

                
04B3:2868:2E38:A3908:62FDA192

                
                

                  
traceroute to 72.134.113.196 (72.134.113.196), 30 hops max, 60 byte packets
 1  a658d6d473ee01ddbe883aba6935df54  0.172 ms
 2  16b034be1fbdf63798891697e6c86336  0.133 ms
 3  c75584d5d5b415b8d2bf57b56931183c  0.403 ms
 4  304ea00828ea9d0a10d47144cca084af  0.787 ms
 5  8263aafac371443e0f92eca52b73c64c  0.339 ms
 6  49f26de3fb706749ff4bac56450935fb  0.694 ms
 7  517d7b8f61d4bc123c4dabd289b97be4  64.287 ms
 8  *
 9  lag-10.asbnva1611w-bcr00.netops.charter.com (66.109.5.116)  84.180 ms
10  lag-24.nwrknjmd67w-bcr00.netops.charter.com (66.109.9.163)  88.925 ms
11  lag-12.nycmny837aw-bcr00.netops.charter.com (66.109.6.27)  88.946 ms
12  lag-22.chcgildt87w-bcr00.netops.charter.com (66.109.6.95)  84.090 ms
13  lag-31.chctilwc00w-bcr00.netops.charter.com (66.109.10.83)  84.179 ms
14  lag-15.lsancarc0yw-bcr00.netops.charter.com (66.109.6.227)  88.862 ms
15  lag-4.lsancarc01r.netops.charter.com (209.18.43.63)  84.101 ms
16  lag-1.lamrcadq01r.netops.charter.com (72.129.9.1)  95.581 ms
17  lag-1.snmncaby01h.netops.charter.com (72.129.10.129)  90.583 ms
18  76.167.31.242 (76.167.31.242)  88.296 ms
19  *
20  *
21  *
22  *
23  *
24  *
25  *
26  *
27  *
28  *
29  *
30  *

                  
PING 72.134.113.196 (72.134.113.196) 56(84) bytes of data.
64 bytes from 72.134.113.196: icmp_seq=1 ttl=47 time=116 ms
64 bytes from 72.134.113.196: icmp_seq=2 ttl=47 time=124 ms
64 bytes from 72.134.113.196: icmp_seq=3 ttl=47 time=107 ms

--- 72.134.113.196 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2002ms
rtt min/avg/max/mdev = 107.528/116.039/124.059/6.757 ms

                  
Start: 2022-08-17T19:19:07-0700
HOST: 30ef5eb1fe5aebdb13f5831c3ccb68cd                            Loss%   Snt   Last   Avg  Best  Wrst StDev
  1.|-- a658d6d473ee01ddbe883aba6935df54                             0.0%    10    0.2   0.2   0.2   0.3   0.0
  2.|-- 2d24c0be8b6863a6b776f81e31f18d36                             0.0%    10    0.4   0.3   0.2   0.5   0.1
  3.|-- 0960dc02df4cc2e2c83a07186203c938                             0.0%    10    0.5   0.5   0.5   0.7   0.1
  4.|-- e2661df0c40ffbccc8713ab28f3ef328                             0.0%    10   36.7   4.4   0.6  36.7  11.3
  5.|-- 8263aafac371443e0f92eca52b73c64c                             0.0%    10    0.4   0.5   0.4   0.6   0.1
  6.|-- b75c459612b29a79b4ed55b2cab2fc2b                             0.0%    10    0.8   1.0   0.7   2.5   0.6
  7.|-- 517d7b8f61d4bc123c4dabd289b97be4                             0.0%    10   74.5  67.4  64.2  84.2   6.7
  8.|-- ???                                                         100.0    10    0.0   0.0   0.0   0.0   0.0
  9.|-- lag-310.asbnva1611w-bcr00.netops.charter.com (209.18.43.58)  0.0%    10   84.0  83.9  83.7  84.1   0.1
 10.|-- lag-13.nwrknjmd67w-bcr00.netops.charter.com (107.14.17.238)  0.0%    10   84.6  90.2  84.0 120.1  12.8
 11.|-- lag-12.nycmny837aw-bcr00.netops.charter.com (66.109.6.27)    0.0%    10   84.3  84.2  83.9  85.2   0.4
 12.|-- lag-22.chcgildt87w-bcr00.netops.charter.com (66.109.6.95)    0.0%    10   84.3  84.4  84.1  85.1   0.3
 13.|-- lag-11.snvacaid17w-bcr00.netops.charter.com (66.109.6.14)    0.0%    10   95.4  92.7  89.5  95.4   2.3
 14.|-- lag-11.snjucacl67w-bcr00.netops.charter.com (66.109.6.8)     0.0%    10   95.0  92.2  89.4  95.0   2.1
 15.|-- lag-13.lsancarc0yw-bcr00.netops.charter.com (66.109.6.7)     0.0%    10   89.4  92.0  89.2  95.0   2.2
 16.|-- lag-4.lsancarc01r.netops.charter.com (209.18.43.63)          0.0%    10   96.4 144.4  88.9 389.6 101.0
 17.|-- lag-1.lamrcadq01r.netops.charter.com (72.129.9.1)            0.0%    10   97.3  97.2  94.4 100.3   2.3
 18.|-- lag-1.snmncaby01h.netops.charter.com (72.129.10.129)         0.0%    10   94.9  95.2  94.7  95.6   0.3
 19.|-- 76.167.31.242                                                0.0%    10   93.3  93.3  93.2  93.8   0.2
 20.|-- cpe-72-134-113-196.natsow.res.rr.com (72.134.113.196)        0.0%    10  113.1 110.9 102.7 118.0   5.1

                  
196.113.134.72.in-addr.arpa domain name pointer cpe-72-134-113-196.natsow.res.rr.com.

**Comment by jfullstackdev** [comment]
is this fixed by now ?

**Comment by airtower-luna** [comment]
> It says post the results in support, and this category seemed like the best fit, so I don't know if its in the right place, but here the results:

This isn't support, this is the community forum. To reach GitHub support, use their support form: https://support.github.com/request

Another option (both is good!) is to contact your ISP and ask why their route to GitHub is slow.

**Comment by iAliJ** [comment]
This post came up on google when I was searching for solution, using a VPN helped me, as mentioned above me it seem like an ISP routing issue. I used to Max 1Mib/s then drops to 200 Kib/s after using a VPN the download is stable at 9 Mib/s

**Comment by ell1e** [comment]
Not sure if this is related, but GitHub's HTTPS clone server handling seems to generally have some networking config issues: https://github.com/orgs/community/discussions/135808 (while e.g. the regular site HTTPS downloads that don't go through the git backend directly are fine)

**Comment by adi-works** [comment]
I tried cloning with both HTTPS and SSH and both are super slow to download. Sometimes it'd even drop the connection.

**Comment by ibizoobysa** [comment]
I’ve totally been there with the painfully slow clone speeds, and it’s super frustrating. For me, using a VPN made a huge difference—my speed went from a snail's pace to something much better. Turns out it was an ISP issue, just like others have pointed out.

**Reply by ell1e** [reply]
If your connection is slow enough, there also seems to be a GitHub-side issue on top, sadly. It seems to use too short timeouts where a progressing but slow git clone can get shot down. At least that's my best guess for what's going on.
